openhands/integrations/provider.py

```diff
@@ -128,17 +145,29 @@ async def _get_latest_provider_token(
 
     async def get_repositories(self, sort: str, app_mode: AppMode) -> list[Repository]:
         """
-        Get repositories from providers
+        Get repositories from providers, including local git repositories if WORKSPACE_BASE is set
         """
 
         all_repos: list[Repository] = []
+        
+        # Get repositories from configured providers
         for provider in self.provider_tokens:
             try:
                 service = self._get_service(provider)
                 service_repos = await service.get_repositories(sort, app_mode)
                 all_repos.extend(service_repos)
             except Exception as e:
                 logger.warning(f'Error fetching repos from {provider}: {e}')
+        
+        # Check for local git repositories if WORKSPACE_BASE is set
+        if os.environ.get('WORKSPACE_BASE'):
```

Comment:
@openhands instead of checking the environment variable, check the relevant config key, which gets set from the env var